packages/mdx-components

jharvey10 opened the pull request:
Package containing all components that can be used in MDX on Platform.

To start, keeping changes self-contained and only porting over a couple components. No changes to the web-app are included in this PR.

There are also some housekeeping-related build/module/typescript changes in this PR.

Closes #1104
Closes #1102 
Relates to #1103 

TODO:
- [x] Package docs

jharvey10 committed:
fix: baseline module updates

preparing for new package

jharvey10 committed:
feat(mdx-components): add mdx-components package

jharvey10 committed:
feat(mdx-components): add storybook and 2 components

- improve linting scripts

jharvey10 committed:
deps: rebuild lockfile and update deps minor/patch

jharvey10 committed:
fix(mdx-components): storybook config

jharvey10 committed:
fix(web-app): sass linter error

jharvey10 committed:
fix(mdx-components): anchor-links styles

francinelucca commented on packages/mdx-components/README.md:
```diff
@@ -0,0 +1,3 @@
+# @carbon-platform/mdx-components
+
+Docs are located [here](../../docs/packages-mdx-components.md).
```
should we create these docs even if empty? or add a TODO/create a ticket for this?

jdharvey-ibm replied:
Yep, I have a todo to add that in this PR. Probably working on that this afternoon.

francinelucca commented on packages/mdx-components/src/main/color-block/color-block.stories.jsx:
```diff
@@ -0,0 +1,16 @@
+/*
```
I think we talked about this, but what was the purpose of this SB? 😅. I think there's a lot this one's missing compared to the one we have on the web-app:
<img width="898" alt="image" src="https://user-images.githubusercontent.com/40550942/183942248-01ecb3cc-f23b-451d-a15a-3b2bce272cc0.png">
VS
<img width="1561" alt="image" src="https://user-images.githubusercontent.com/40550942/183942322-90700d6e-a329-4c06-8c35-3b84a2b9a090.png">

jdharvey-ibm replied:
Fair question! A few purposes:

1. New home for our existing stories that apply to mdx components
2. Aids in mdx-component development (no need to have the web-app up and running)
3. Enables eventual snapshot testing specifically for the mdx-components to prevent regression

jdharvey-ibm replied:
Updated to match Alison's recommendations. Info now has parity with the one in the web-app.

francinelucca reviewed (commented):
some comms, setup LGTM

alisonjoseph commented:
I'm assuming the mdx components will eventually move out of the web app storybook and into this one? If so, can we match the storybook config options?
- Docs should be the default ab
- Code shown by default
- Active controls on docs page under the code to control the component
- remove Controls/Actions tabs from the Canvas view

alisonjoseph commented on packages/mdx-components/src/main/anchor-links/anchor-links.stories.jsx:
```diff
@@ -0,0 +1,37 @@
+      <AnchorLink>Link 6</AnchorLink>
+      <AnchorLink>Link 7</AnchorLink>
+    </AnchorLinks>
+    <div>is it red?</div>
```
whats this for?

alisonjoseph commented on packages/mdx-components/src/main/anchor-links/anchor-links.stories.jsx:
```diff
@@ -0,0 +1,37 @@
+      <AnchorLink>Link 6</AnchorLink>
+      <AnchorLink>Link 7</AnchorLink>
+    </AnchorLinks>
+    <div>is it red?</div>
```
```suggestion
    <div>is it red?</div>
```
What is this for?

jdharvey-ibm replied:
Testing that was supposed to be removed 😅

jharvey10 committed:
fix(mdx-components): code review updates

jharvey10 committed:
fix(mdx-components): update storybook

- also fix a style bug with anchor links

jharvey10 committed:
fix(web-app): apply first mdx-components

alisonjoseph reviewed (approved):
Storybook LGTM! 🥳 

Is there a separate issue to port over the rest of the components? (I can help out!)

francinelucca commented:
@alisonjoseph I think that'd be #1103

francinelucca commented on services/web-app/components/mdx/components.js:
```diff
@@ -73,26 +62,26 @@ const components = {
-  UnorderedList,
-  ListItem,
-  Tag,
+  // Button,
+  // StructuredListWrapper,
+  // StructuredListHead,
+  // StructuredListRow,
+  // StructuredListCell,
+  // StructuredListBody,
+  // UnorderedList,
+  // ListItem,
+  // Tag,
```
can we delete instead?
```suggestion
```

jdharvey-ibm replied:
Yeah, definitely. I'll make that change.

francinelucca commented on services/web-app/components/mdx/components.js:
```diff
@@ -73,26 +62,26 @@ const components = {
   AccordionItem,
-  AnchorLink,
-  AnchorLinks,
+  // AnchorLink,
+  // AnchorLinks,
```
```suggestion
```

francinelucca commented on services/web-app/components/mdx/components.js:
```diff
@@ -73,26 +62,26 @@ const components = {
   Caption,
   CardGroup,
-  ColorBlock,
+  // ColorBlock,
```
```suggestion
```

francinelucca reviewed (approved):
LGTM!
is there a way to make sonarcloud ignore this package so it doesn't make testing fail?

jharvey10 committed:
test(mdx-components): test infrastructure and basic snapshot testing

- also adds additional unit test for get-run-mode
- adds title to color block for accessibility and RTL testing

jharvey10 committed:
refactor(web-app): remove commented out lines from components.js

jharvey10 committed:
Merge branch 'main' of github.com:carbon-design-system/carbon-platform into feat/mdx-components

jharvey10 committed:
deps: update lock file

jharvey10 committed:
fix(web-app): linter error in page-tabs scss

jharvey10 committed:
docs(mdx-components): add docs for how to use the package

jharvey10 committed:
fix(web-app): change anchor link import location to new package
